File: lib/models/consulta_model.php
class Consulta {
public $pet, $id, $titulo, $data;

function __construct($id, $pet, $titulo, $data) {
$this->id = $id;
$this->pet = $pet;
$this->titulo = $titulo;
$this->data = $data;
}

function toJson() {
return [
'id' => rand(0, 1000),
'pet' => $this->pet,
'titulo' => $this->titulo,
'data' => $this->data
];
}

function fromJson($map) {
$this->id = $map["id"];
$this->titulo = $map["titulo"];
$this->pet = $map["pet"];
$this->data = $map["data"];
}
}

File: lib/models/consulta_model.dart
import 'dart:math';

class Consulta {
  late String pet, id, titulo, data;

  Consulta({required this.id, required this.pet, required this.titulo,required  this.data});

  Random random = new Random();

  Map<String, Object> toJson() {
    return {
      'id': random.nextInt(1000).toString(),
      'pet': pet,
      'titulo': titulo,
      'data': data
    };
  }

  Consulta.fromJson(dynamic map) {
    id = map["id"];
    titulo = map["titulo"];
    pet = map["pet"];
    data = map["data"];
  }
}
